README.md

```diff
@@ -1,5 +1,232 @@
-# iota-sdk
+# IOTA SDK Library
 
 [![Coverage Status](https://coveralls.io/repos/github/iotaledger/iota-sdk/badge.svg?branch=develop)](https://coveralls.io/github/iotaledger/iota-sdk?branch=develop)
 
-WIP
+The IOTA SDK is a Rust-based project that provides a convenient and efficient way to interact with Shimmer nodes in the
+Shimmer network. It consists of two main modules: `wallet` and `client`.
+
+The `wallet` module is stateful, with a standardized interface for developers to build applications involving value
+transactions. It uses high-level functions that simplify everyday operations. It can optionally interact
+with [IOTA Stronghold](https://github.com/iotaledger/stronghold.rs/) for seed handling and storage and state backup.
+
+The `client` module is stateless. It aims to provide more flexibility and access to low-level functions.
+
+## Table of Contents
+
+- [Requirements](#requirements)
+  - [Dependencies](#dependencies)
+- [Getting Started](#getting-started)
+  - [Install the IOTA SDK](#install-the-iota-sdk)
+  - [Usage](#usage)
+    - [Wallet](#wallet)
+    - [Client](#client)
+    - [Examples](#examples)
+  - [API Reference](#api-reference)
+- [Contribute](#contribute)
+- [License](#license)
+- [Acknowledgments](#acknowledgments)
+
+## Features
+
+- **Wallet module**: The `wallet` module in the IOTA SDK provides high-level functions for managing accounts, generating
+  addresses, creating transactions, and interacting with the Shimmer network. It offers a user-friendly interface for
+  developers to build applications on the Shimmer network.
+
+- **Client module**: The `client` module in the IOTA SDK offers low-level functions that allow you to have
+  fine-grained control over their interactions with Shimmer nodes. It provides access to the underlying API endpoints
+  and enables advanced operations such as custom message construction and direct communication with the network.
+
+- **Python, Node.js, and WASM Bindings**: The IOTA SDK includes bindings for Python, Node.js, and WASM, which allow you to use the
+  SDK in your preferred programming language. These bindings provide seamless integration with existing Python and
+  Node.js projects, enabling cross-platform compatibility and flexibility.
+
+## Branching Structure for Development
+
+This library follows the following branching strategy:
+
+| Branch       | Description                                                                                                                    |
+|--------------|--------------------------------------------------------------------------------------------------------------------------------|
+| `develop`    | Ongoing development for future releases of the networks. This branch gets merged into `staging` on releases.                   |
```

Comment:
maybe change "networks" to "staging network", to show better which network it is for